README.md

```diff
@@ -242,3 +244,81 @@ const result = await myConsumer.handlerSpy.waitForMessageWithId('1')
 expect(result.processingResult).toEqual('consumed')
 ```
 
+## Payload Offloading
+Payload offloading allows you to manage large message payloads by storing them in external storage, bypassing any message size restrictions imposed by queue systems.
+
+### Configuration
+
+1. **Install a payload store implementation (S3 in example):**
+   ```bash
+   npm install @message-queue-toolkit/s3-payload-store
+   ```
+
+2. **Configure your setup:**
+    ```typescript
+    import {S3} from '@aws-sdk/client-s3'
+    import {S3PayloadStore} from '@message-queue-toolkit/s3-payload-store'
+    import {PayloadStoreConfig} from "@message-queue-toolkit/core"
+    
+    const s3Client = new S3({
+        region: 'your-s3-region',
+        credentials: {
+            accessKeyId: 'your-access-key-id',
+            secretAccessKey: 'your-secret-access-key'
+        }
+    })
+    
+    const payloadStoreConfig = {
+        store: new S3PayloadStore(
+            {s3: s3Client},
+            {
+                bucketName: 'your-s3-bucket-name',
+                keyPrefix: 'optional-key-prefix'
+            }
+        ),
+        messageSizeThreshold: 256 * 1024 // 256KB
+    }
+    
+    export class MyConsumer extends AbstractSqsConsumer<> {
+        constructor(
+            // dependencies and options
+        ) {
+            super(dependencies, {
+                // rest of the configuration
+                payloadStoreConfig,
+            })
+        }
+    }
+    
+    export class MyPublisher extends AbstractSqsPublisher<> {
+        constructor(
+            // dependencies and options
+        ) {
+            super(dependencies, {
+                // rest of the configuration
+                payloadStoreConfig,
+            })
+        }
+    }
+    ```
+3. **Usage:**
+   Messages larger than the `messageSizeThreshold` are automatically offloaded to the configured store (S3 bucket in example). 
+   The queue manages only references to these payloads, ensuring compliance with size limits.
+4. **Managing payload lifecycle:**
+   It is important to note that once messages are processed, their payloads are not automatically deleted from the external storage.
```

Comment:
The primary reason is simplification. 
Managing payloads becomes complicated with fan-out, as we need to track if all consumers have read the message, which is a challenging task. 

Additionally, you'll likely need to set up some expiration policy for unconsumed or DLQ-rerouted messages since you can't keep them indefinitely. This leads to having an expiration policy in place anyway.

Reply:
Fair point. Maybe we should mention that in the documentation as well?

Reply:
Expanded section with the extra note: https://github.com/kibertoad/message-queue-toolkit/pull/167/commits/d3224e9fb3812cfd4a6ca230955b15c43d0d9579